src/batched/KokkosBatched_QR_WithColumnPivoting_TeamVector_Internal.hpp

```diff
@@ -96,6 +96,7 @@ namespace KokkosBatched {
 				    A, as0, as1,
 				    A, as0, as1,
 				    norm, 1);
+	member.team_barrier();
```

Comment:
You can probably get away with not including this team_barrier(), since the PerThread call made after a TeamVector reduction will likely not use the result until quite a large number of commands later (but no promise).